src/webgl-preview.ts

```diff
@@ -310,17 +323,54 @@ export class WebGLPreview {
     }
 
     if (this.renderTubes) {
-      console.warn('Volumetric rendering is experimental and may not work as expected or change in the future.');
+      console.warn('Volumetric rendering is experimental. It may not work as expected or change in the future.');
       const light = new AmbientLight(0xcccccc, 0.3 * Math.PI);
       // threejs assumes meters but we use mm. So we need to scale the decay of the light
       const dLight = new PointLight(0xffffff, Math.PI, undefined, 1 / 1000);
       dLight.position.set(0, 500, 500);
       this.scene.add(light);
       this.scene.add(dLight);
     }
+  }
+
+  createGroup(name: string): Group {
+    const group = new Group();
+    group.name = name;
+    group.quaternion.setFromEuler(new Euler(-Math.PI / 2, 0, 0));
+    if (this.buildVolume) {
+      group.position.set(-this.buildVolume.x / 2, 0, this.buildVolume.y / 2);
+    } else {
+      // FIXME: this is just a very crude approximation for centering
+      group.position.set(-100, 0, 100);
+    }
+    return group;
+  }
+
+  // Render the next increment. This is done by removing the previous layer and rendering all layers up to the current one.
+  // This way it preserves the state of the previous layers but it does render the top layer again b/c it probably wasnt completed before.
+  renderInc(): void {
```

Comment:
If I understand correctly, the role of this method is to do a render based on state changes and the key is to use `this.prevLayerIndex` to only render what's new (and backtrack just a little in case of incomplete layers). 

There's a part I'm slightly confused. The only call is commented out, so it's technically unused.

Reply:
I also think that the naming should tell more about that story. `renderNextIncrement` is more explicit and makes the first part of the code comment obsolete. And in the context of a library user, what is an increment? What would they understand of this API, if they have zero knowledge on how it's implemented?

Reply:
I started out the PR with this method but in the end I came up with a different solution. This is where most of my confusion some from: I still have renderInc in head as better way to do well, incremental processing. The difference is in how the parser is used. So renderInc would be used when new chunks of gcode would be parsed, processed and rendered. But for loading files in the demo, and to maintain the gradient, I needed to parse the file first.

So technically this method should be removed now. Agreed. We can always dig it up later if we need it.